src/einsteinpy/bodies.py

```diff
@@ -15,195 +15,192 @@
 Data references can be found in :py:mod:`~einsteinpy.constant`
 """
 
+from dataclasses import dataclass
+from textwrap import dedent
+
 from astropy import units as u
 
-from einsteinpy import constant
+from einsteinpy.constant import R_sun, Solar_Mass
 
 __all__ = ["Body"]
 
 
+@dataclass
 class Body:
     """
     Class to create a generic Body
+
+    Parameters
+    ----------
+    name : string
+        Name or ID of the body
+    mass : ~astropy.units.kg
+        Mass of the body
+    q : ~astropy.units.C, optional
+        Charge on the body
+    R : ~astropy.units
+        Radius of the body
+    differential : ~einsteinpy.coordinates.differential.*, optional
+        Complete coordinates of the body
+    parent : Body, optional
+        The parent object of the body
+        Useful in case of multibody systems
     """
 
-    @u.quantity_input(mass=u.kg, R=u.km)
-    def __init__(
-        self,
-        name="Generic Body",
-        mass=0 * u.kg,
-        q=0 * u.C,
-        R=0 * u.km,
-        differential=None,
-        parent=None,
-    ):
-        """
-        Parameters
-        ----------
-        name : string
-            Name or ID of the body
-        mass : ~astropy.units.kg
-            Mass of the body
-        q : ~astropy.units.C, optional
-            Charge on the body
-        R : ~astropy.units
-            Radius of the body
-        differential : ~einsteinpy.coordinates.differential.*, optional
-            Complete coordinates of the body
-        parent : Body, optional
-            The parent object of the body
-            Useful in case of multibody systems
-        """
-        if differential:
-            if differential.system == "Cartesian":
-                self.pos_vec = [differential.x, differential.y, differential.z]
-                self.vel_vec = [differential.v_x, differential.v_y, differential.v_z]
+    name: str = "Generic Body"
+    mass: u.kg = 0 * u.kg
+    q: u.C = 0 * u.C
+    R: u.km = 0 * u.km
+    coords = None

```

Comment:
This will need a type hint, otherwise `coords` won't be considered an instance variable.
```py
from typing import Union
from einsteinpy.coordinates.differential import CartesianDifferential, SphericalDifferential, BoyerLindquistDifferential

# Other code

coords: Union[CartesianDifferential, SphericalDifferential, BoyerLindquistDifferential, None] = None
parent: Union["Body", None] = None
```

Reply:
Sorry, got it

Reply:
Is this added?

Reply:
Sorry, thought I'd added it, now it's there